ibr/ibr-sdk/src/renderer.js

```diff
@@ -10,49 +10,59 @@
-
-    // Put coordinates in data structure
-    var coords = decodedCoordsString.split(",");
-    var decodedCoordsList = [];
-    for (const coord of coords) {
-        var coordinate = coord.split(" ");
-        if (coordinate.length === 2) {
-            decodedCoordsList.push( coordinate );
-        }
+    // Decode Coordinates from data.coordinates_lookup
+    var coordsLookup = deserializedData.coordinates_lookup;
+    var coordsLookupBuffer = coordsLookup.buffer.slice(coordsLookup.byteOffset, coordsLookup.byteOffset+coordsLookup.length);
+    var coordsLookupDV = new DataView(coordsLookupBuffer);
+    var coordsLookupList = [];
+    for (var i = 0; i < coordsLookup.length/4; i+=4) {
+        coordsLookupList.push(coordsLookupDV.getFloat32(i, false));
     }
 
     // Read multiple ranges from Visualization.coordinates array and store them in sessionStorage for visualization later
-    var layerCoordinates = [];
-    for (var i = 0; i < coordsRange.length; i += 2) {
-        layerCoordinates[i/2] = [];
-        for (var x = coordsRange[i]; x < coordsRange[i+1]; x++) {
-            layerCoordinates[i/2].push( decodedCoordsList[x] );
+    var layerCoordinates = [], layerPH = [], coordsLine = [];
+    for (const coordsRangeItem of coordsRangeList) {
+        layerPH = [];
+        for (var i = 0; i < coordsRangeItem.length; i+=2) {
+            coordsLine = [];
+            for (var j = coordsRangeItem[i]; j <= coordsRangeItem[i+1]; j+=3) {
+                coordsLine.push(coordsLookupList[j]);
+                coordsLine.push(coordsLookupList[j+1]);
+                coordsLine.push(coordsLookupList[j+2]);
+            }
+            layerPH.push(coordsLine);
         }
+        layerCoordinates.push(layerPH);
     }
 
-    // Construct three.js objects from decoded coordinates
-    var material1 = new THREE.LineBasicMaterial( { color: 0x0000ff } );
+    // Render data into three.js objects
+    var materials = [new THREE.LineBasicMaterial( { color: 0x00ffff } ), new THREE.LineBasicMaterial( { color: 0xff0000 } )];
     var points = [], lines = [];
     var geometry, line;
-    for (const layer of layerCoordinates) {
-        for (const coordinate of layer) {
-            points.push( new THREE.Vector3( coordinate[0], coordinate[1], 0 ) );
+    for (var i = 0; i < layerCoordinates.length; i++) {
+        for (const l of layerCoordinates[i]) {
+            points = [];
+            for (var j = 0; j < l.length; j+=3) {
+                points.push( new THREE.Vector3( l[j], l[j+1], l[j+2] ) );
+            }
+            geometry = new THREE.BufferGeometry().setFromPoints( points );
+            line = new THREE.LineLoop( geometry, materials[i] );
+            lines.push( line );
         }
-        geometry = new THREE.BufferGeometry().setFromPoints( points );
-        line = new THREE.LineLoop( geometry, material1 );
-        lines.push( line );
     }
     return lines;
```

Comment:
so right now we have two types of lines, the boundary and the wall layer, and it isn't super obvious right now.

As this set of supported elements gets larger and larger it's going to be difficult for the reader to parse out what is what in just an array of THREE.Lines

In the next change, let's start building custom render functions for each thing so we can isolate specific elements to make the code easier to maintain and debug.